### PR TITLE
Add ServersController#duplicate action

### PR DESCRIPTION
**Pull request by B-Rass**

### CODE REVIEW

**Comment by B-Rass** [comment]
@nicolas-brousse maybe I could add a test on the new duplicate method  😅

File: spec/requests/servers_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+      before { get server_path(server.name) }
+
+      it { expect(response).to render_template(:show) }
+    end
+
+    context "while using serial number" do
+      before { get server_path(server.numero) }
+
+      it { expect(response).to render_template(:show) }
+    end
```
Comment (nicolas-brousse) [inline spec/requests/servers_request_spec.rb:48]: Maybe add a test for 404

File: spec/requests/servers_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+      it "creates a new Server" do
+        expect do
+          post servers_path, params: { server: valid_attributes }
+        end.to change(Server, :count).by(1)
+      end
+
+      it "redirects to the created server" do
+        post servers_path, params: { server: valid_attributes }
+        expect(response).to redirect_to(assigns(:server))
```
Comment (nicolas-brousse) [inline spec/requests/servers_request_spec.rb:79]: Prefer use of exact url helper instead

File: spec/requests/servers_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+          .to raise_error(ActionController::ParameterMissing)
+      end
+
+      it "does not create a new Server with an existing serial number" do
+        expect do
+          post servers_path, params: { server: server2.attributes.except(["id"]) }
+        end.not_to change(Server, :count)
+      end
+
+      it "does not create a new Server with invalid parameters", :aggregate_failures do
```
Comment (nicolas-brousse) [inline spec/requests/servers_request_spec.rb:102]:
```suggestion
      it "does not create a new Server with invalid parameters" do
```
It seems not to be necessary here

File: spec/requests/servers_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+    it :aggregate_failures do
+      expect do
+        expect do
+          expect do
+            post import_servers_path, params: { import: { file: csv,
+                                                          room_id: Room.first.id,
+                                                          server_state_id: ServerState.first.id } }
+          end.to change(Server, :count).by(26)
+        end.to change(Frame, :count)
+      end.to change(Bay, :count)
```
Comment (nicolas-brousse) [inline spec/requests/servers_request_spec.rb:247]: If possible try to move this in on block only and use `#and` matcher https://rspec.info/features/3-12/rspec-expectations/compound-expectations/

**Review by nicolas-brousse — CHANGES_REQUESTED** [review]

**Review by nicolas-brousse — APPROVED** [review]
